### ISSUE TITLE
terminal options crashes

### ISSUE THREAD

**Issue by joecar77** [mintty/mintty#1295]
mintty 3.7.6 running on Windows 10.

From mintty title bar, click the menu button, click Options, click Terminal.

mintty then stops responding (verified with Task Manager).

<img width="625" alt="mintty options terminal not responding" src="https://github.com/user-attachments/assets/939fef9d-cad0-47e2-941e-ae543e4e520e">

**Comment by mintty** [comment]
I do not reproduce this. Also tried in Git Bash which I see you used.
Please provide information about your environment. Upload your mintty config file, `uname -a`, and any unusual configuration you may have applied to Git Bash.

**Comment by mintty** [comment]
And also the list of printers configured in Windows. Are there any remote printers?

**Comment by joecar77** [comment]
```
$ cat ~/.minttyrc
BackgroundColour=22,0,45
Transparency=medium
CursorType=block
Font=Lucida Console
RightClickAction=paste
Columns=122
Rows=52
PgUpDnScroll=yes
CursorColour=0,255,0
ForegroundColour=239,239,239
Printer=Microsoft Print to PDF
ThemeFile=
FontHeight=8
ClicksPlaceCursor=no
CopyTab=yes
CopyAsHTML=no
ScrollbackLines=20000
OpaqueWhenFocused=no
```

**Comment by joecar77** [comment]
I have no printers configured in Windows (other than Microsoft Print to PDF).

**Comment by joecar77** [comment]
```
$ cat ~/.bashrc
#!/usr/bin/bash

alias h='history'
alias hh='history | grep --'

alias grep='grep --color=auto'
alias egrep='egrep --color=auto'
alias fgrep='fgrep --color=auto'

alias rm='rm -i'
alias cp='cp -i'
alias mv='mv -i'

alias ls='ls -F --color=auto'
alias ll='ls -al'
alias la='ls -A'
alias l='ls -C'

alias tree='tree -aFC'

source /mingw64/share/git/completion/git-completion.bash
source /mingw64/share/git/completion/git-prompt.sh

export PS1='\[\e[1;35m\]\u\[\e[33m\]@\[\e[1;36m\]\h\[\e[33m\]:\[\e[1;31m\]\w\[\e[32m\]$(__git_ps1 "[%s]")\[\e[33m\]#\[\e[0m\] '

```

**Comment by mintty** [comment]
Please run `MINTTY_DEBUG=o mintty` from another mintty session and click the Terminal section in Options. How much output do you get?

**Comment by joecar77** [comment]
**MINTTY_DEBUG=o**

<img width="432" alt="image" src="https://github.com/user-attachments/assets/38708e73-f8c5-41be-899d-3bf66c634cdf">

```
Joe@syn0103:~$ MINTTY_DEBUG=o mintty
WM_NOTIFY
WM_NOTIFY: GetSelection
WM_NOTIFY: Destroy
WM_NOTIFY: cleanup
WM_NOTIFY: create
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_NOTIFY: refresh
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_USER
WM_USER: lookup
WM_USER: handler
WM_USER: end
WM_NOTIFY
WM_NOTIFY: GetSelection
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_NOTIFY: Destroy
WM_NOTIFY: cleanup
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_NOTIFY: create
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
Joe@syn0103:~$
```

**Comment by mintty** [comment]
Can you build mintty yourself? It's easy, just packages gcc-core and make needed.
What if you add the following to file printers.c, line 16:
```
printer_start_enum(void)
{
return 0;

```

**Comment by joecar77** [comment]
I installed git-scm... I didn't install cygwin...

**Comment by mintty** [comment]
As I don't reproduce the issue and I'm currently quite clueless, your cooperation to debug further will be much appreciated.
You could for example also install the Git for Windows SDK (download link at bottom of https://gitforwindows.org/) or plain MSYS2 which is the base for Git for Windows.

**Comment by mintty** [comment]
Please try the attached debug build: 
[mintty-debug.zip](https://github.com/user-attachments/files/17634541/mintty-debug.zip)

**Comment by joecar77** [comment]
I understand. I have limited time, but I will schedule the debug build. Thanks.

**Comment by joecar77** [comment]
( Thanks, I do appreciate your time )

**Comment by mintty** [comment]
> I probably have to kill all instances of mintty in order to try the debug build...?

No, just run this instance like above, setting MINTTY_DEBUG=o.

**Comment by joecar77** [comment]
<img width="404" alt="image" src="https://github.com/user-attachments/assets/b2f7b94d-df6f-4ec6-9580-146527a37f31">

**Comment by mintty** [comment]
You need to run it from inside your Git-for-Windows environment, from another terminal session, as before.
You can check the library resolution with `ldd mintty.exe`.

**Comment by joecar77** [comment]
ah, got it...

![image](https://github.com/user-attachments/assets/9f9479c1-56e1-47f8-8d3a-bc2b493e5548)


```
Joe@syn0103:~/Downloads/mintty-debug# MINTTY_DEBUG=o ./mintty.exe
WM_NOTIFY
WM_NOTIFY: GetSelection
WM_NOTIFY: Destroy
WM_NOTIFY: cleanup
WM_NOTIFY: create
[2214 ev 0] type 3 (1 1 0) label About... col 0
[2214 ev 0] type 3 (1 1 0) label Save col 2
[2214 ev 0] type 3 (1 1 0) label Cancel col 3
[2214 ev 0] type 3 (1 1 0) label Apply col 4
[2214 ev 0] type 3 (1 1 1) label &Foreground... col 0
[2214 ev 0] type 3 (1 1 1) label &Background... col 1
[2214 ev 0] type 3 (1 1 1) label &Cursor... col 2
[2214 ev 0] type 0 (1 1 1) label &Theme col 131072
[2214 ev 0] type 3 (1 1 1) label Color Scheme Designer col 0
[2214 ev 0] type 3 (1 1 0) label Store col 1
[2214 ev 0] type 1 (1 1 1) label (null) col 0
[2214 ev 0] type 2 (1 1 1) label Opa&que when focused col 0
[2214 ev 0] type 0 (1 1 0) label (null) col 2
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
[1883 ev 2] type 0 (1 1 0) label (null) col 2
WM_COMMAND: handle
WM_COMMAND: end
[2214 ev 0] type 3 (1 1 1) label ◄ col 1
[2214 ev 0] type 3 (1 1 1) label ► col 3
[2214 ev 0] type 1 (1 1 1) label (null) col 0
[2214 ev 0] type 2 (1 1 1) label Blinkin&g col 0
WM_NOTIFY: refresh
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_USER
WM_USER: lookup
[682 ev 6] type 1 (1 1 1) label (null) col 0
WM_USER: handler
WM_USER: end
WM_NOTIFY
WM_NOTIFY: GetSelection
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_NOTIFY: Destroy
WM_NOTIFY: cleanup
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_NOTIFY: create
[2214 ev 0] type 3 (1 1 0) label About... col 0
[2214 ev 0] type 3 (1 1 0) label Save col 2
[2214 ev 0] type 3 (1 1 0) label Cancel col 3
[2214 ev 0] type 3 (1 1 0) label Apply col 4
[2214 ev 0] type 0 (1 1 0) label &Type col 0
[2214 ev 0] type 0 (1 1 1) label &Answerback col 1
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
[1883 ev 2] type 0 (1 1 1) label &Answerback col 1
WM_COMMAND: handle
WM_COMMAND: end
[2214 ev 0] type 0 (1 1 0) label (null) col 0
[2214 ev 0] type 3 (1 1 0) label ► &Play col 1
[2214 ev 0] type 0 (1 0 1) label &Wave col 0
[2214 ev 0] type 2 (1 0 1) label &Flash col 0
[2214 ev 0] type 2 (1 0 1) label &Highlight in taskbar col 1
[2214 ev 0] type 2 (1 0 1) label &Popup col 2
[2214 ev 0] type 0 (1 1 0) label (null) col 0


```

**Comment by mintty** [comment]
Thanks; as I had suspected before, it stops right before (or while) setting up the printer selection.
Please run the following test program, just listing the configured printers (derived from mintty file printers.c):
[printers.zip](https://github.com/user-attachments/files/17636688/printers.zip)

**Comment by joecar77** [comment]
printers.exe just hangs... I had to kill it after a minute.

```
Joe@syn0103:~/Downloads/printers# ./printers.exe

Joe@syn0103:~/Downloads/printers#
```

**Comment by mintty** [comment]
So as I had suspected, printers enumeration blocks, which is also described to be possible for function EnumPrinters but had never occurred so far, or been reported. I think I'll change to retrieve the printers list from the registry then.

**Comment by joecar77** [comment]
Thanks for looking at it :^)

**Comment by BrianInglis** [comment]
*Cygwin* profiles use:
```
$ grep -iC1 PRINTER /etc/profile.d/cygwin.sh

  # Define default printer
  p='/proc/registry/HKEY_CURRENT_USER/Software/Microsoft/Windows NT/CurrentVersion/Windows/Device'
  if [ -e "${p}" ] ; then
    read -r PRINTER < "${p}"
    PRINTER=${PRINTER%%,*}
  fi
```
Enumerate (above default is just concatenation of name comma device settings):
```
$ regtool list -v /proc/registry/HKEY_CURRENT_USER/Software/Microsoft/Windows\ NT/CurrentVersion/Devices/
Microsoft XPS Document Writer (REG_SZ) = "winspool,Ne00:"
Microsoft Print to PDF (REG_SZ) = "winspool,Ne01:"
EPSON WF-3720 Series (REG_SZ) = "winspool,Ne07:"
EPSONWF-3720 (WF-3720 Series) (REG_SZ) = "winspool,Ne03:"
Generic / Text Only (REG_SZ) = "winspool,FILE:"
```

**Comment by joecar77** [comment]
```
$ grep -i PRINTER /etc/profile.d/*.sh
$
```

**Comment by mintty** [comment]
> "HKEY_CURRENT_USER/Software/Microsoft/Windows NT/CurrentVersion/Devices"

I've uploaded a patch to list printers from the registry, using the above entry. But I think I'll switch to
HKEY_CURRENT_USER/SOFTWARE/Microsoft/Windows NT/CurrentVersion/PrinterPorts
which sometimes contains fewer weird redundant entries.
And there's yet another one:
HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Windows NT/CurrentVersion/Print/Printers
which some applications seem to use,
with attributes "Name" (mostly identical with the key name) and sometimes "PerUserName" which overrides Name if set and nonempty - what a mess.

**Comment by BrianInglis** [comment]
Your weird "redundant" entries may be some user's *essential* output device!
If the differences are undocumented, maybe get them all in one request, and merge them all into a list of unique entries so no entries get missed? 
Ask on the apps list about printer reg keys and entries?

**Comment by mintty** [comment]
On one of my systems, the list in HKEY_CURRENT_USER/SOFTWARE/Microsoft/Windows NT/CurrentVersion/Devices adds two entries to that in .../PrinterPorts: "OneNote" which is a misconfigured alias for my hardware printer, while there are 3 other OneNote entries, and "Microsoft XPS Document Writer" which seems to be an alias of "Microsoft Print to PDF" and isn't listed in the printer menu of any other application. So the PrinterPorts list appears to be more reliable and I should switch to it.

**Comment by mintty** [comment]
Released 3.7.7.
